app/helpers/application_helper.rb

```diff
@@ -1,2 +1,18 @@
 module ApplicationHelper
+  def bootstrap_class_for flash_type
+    { success: "alert-success", error: "alert-danger", alert: "alert-warning", notice: "alert-info" }[flash_type.to_sym] || flash_type.to_s
+  end
+
+  def flash_messages(opts = {})
+    flash.each do |msg_type, message|
+      concat(content_tag(:div, message, class: "alert #{bootstrap_class_for(msg_type)} alert-dismissible", role: 'alert') do
+        concat(content_tag(:button, class: 'close', data: { dismiss: 'alert' }) do
+          concat content_tag(:span, '&times;'.html_safe, 'aria-hidden' => true)
```

Comment:
Re Austin's commit message comment - my understanding is that commit messages are supposed to be quite lengthy. In fact, seantheprogrammer and others have discussed on slack how your commit message should have a title and then at least a paragraph underneath explaining what you actually changed in the code. I like commit messages to be descriptive so that I dont have to scour the code to know exactly what's going on. If I have a good descriptive commit message, then I can read the code looking for what I know is important. I suspect there are different perspectives on this. For the rest of the group project, do you both prefer less descriptive commits? As I said, I prefer reading lengthy ones but am totally fine to go with group consensus here.